Specs/Core/Matrix3Spec.js

```diff
@@ -148,6 +148,24 @@ defineSuite([
         expect(returnedResult).toEqualEpsilon(expected, CesiumMath.EPSILON15);
     });
 
+    it('fromHeadingPitchRoll computed correctly', function() {
+        var sPi = Math.sin(CesiumMath.PI);
+        var cPi = Math.cos(CesiumMath.PI);
+        var sPiOver2 = Math.sin(CesiumMath.PI_OVER_TWO);
+        var cPiOver2 = Math.cos(CesiumMath.PI_OVER_TWO);
+        var s3PiOver2 = Math.sin(CesiumMath.PI_OVER_TWO * 3);
+        var c3PiOver2 = Math.cos(CesiumMath.PI_OVER_TWO * 3);
+
+        var headingPitchRoll = new HeadingPitchRoll(-CesiumMath.PI_OVER_TWO, -CesiumMath.PI, CesiumMath.PI_OVER_TWO * 3);
+        var expected = new Matrix3(cPi * cPiOver2, -c3PiOver2 * sPiOver2 + s3PiOver2 * sPi * cPiOver2, s3PiOver2 * sPiOver2 + c3PiOver2 * sPi * cPiOver2,
```

Comment:
Tests written this way are generally not useful because it just simply copies the same math done in `Matrix3.fromHeadingPitchRoll`.  Most of the time there's a bug, it's because the bad math exists in both cases.

For cases like this, we prefer to use hardcoded  known good values (usually created via a trusted source like STK Components, Matlab, or Wolfram Alpha.  We then put a comment above the hardcoded values indicating where they are from.